validator/src/journal/block_scheduler.rs

```diff
@@ -164,11 +165,23 @@ impl<B: BlockStatusStore> BlockSchedulerState<B> {
                         ready.push(block);
                     }
                 }
-            } else {
+            } else if previous_validity == BlockStatus::Valid {
```

Comment:
I don't want to block this PR but, I don't think this was the original intent for this function. Checking for block status inside the imperative 'schedule' function seems very awkward to me. I think it is the caller's responsibility to know when it is correct to schedule. The function only checks if previous results are available or else it delays scheduling the block. I think it should stay this way. The fact that we need to check for block validation here, hints at missing logic elsewhere. Please, give it a thought.